Sources/MongoSwift/Operations/AggregateOperation.swift

```diff
@@ -85,33 +94,56 @@ internal struct AggregateOperation<CollectionType: Codable, OutputType: Codable>
         let opts = try encodeOptions(options: self.options, session: session)
         let pipeline: BSONDocument = ["pipeline": .array(self.pipeline.map { .document($0) })]
 
-        let result: OpaquePointer = self.collection.withMongocCollection(from: connection) { collPtr in
-            pipeline.withBSONPointer { pipelinePtr in
-                withOptionalBSONPointer(to: opts) { optsPtr in
-                    ReadPreference.withOptionalMongocReadPreference(from: self.options?.readPreference) { rpPtr in
-                        guard let result = mongoc_collection_aggregate(
-                            collPtr,
-                            MONGOC_QUERY_NONE,
-                            pipelinePtr,
-                            optsPtr,
-                            rpPtr
-                        ) else {
-                            fatalError(failedToRetrieveCursorMessage)
+        return try pipeline.withBSONPointer { pipelinePtr in
+            try withOptionalBSONPointer(to: opts) { optsPtr in
+                try ReadPreference.withOptionalMongocReadPreference(from: self.options?.readPreference) { rpPtr ->
+                    MongoCursor<OutputType> in
+                    var result: OpaquePointer
+                    var client: MongoClient
+                    var decoder: BSONDecoder
+                    switch self.target {
+                    case let .database(db):
+                        client = db._client
+                        decoder = db.decoder
+                        result = db.withMongocDatabase(from: connection) { dbPtr in
+                            guard let result = mongoc_database_aggregate(
+                                dbPtr,
+                                pipelinePtr,
+                                optsPtr,
+                                rpPtr
+                            ) else {
+                                fatalError(failedToRetrieveCursorMessage)
+                            }
+                            return result
+                        }
+                    case let .collection(coll):
+                        client = coll._client
+                        decoder = coll.decoder
+                        result = coll.withMongocCollection(from: connection) { collPtr in
+                            guard let result = mongoc_collection_aggregate(
+                                collPtr,
+                                MONGOC_QUERY_NONE,
+                                pipelinePtr,
+                                optsPtr,
+                                rpPtr
+                            ) else {
+                                fatalError(failedToRetrieveCursorMessage)
+                            }
+                            return result
                         }
-                        return result
                     }
+
+                    // since mongoc_collection_aggregate and mongoc_database_aggregate don't do any I/O, use forceIO
```

Comment:
I think this comment is out of date and can be removed.

Reply:
Ooh oops, @patrickfreed what was it doing before? what is forceIO?

Reply:
A lot of mongoc functions that return cursors are lazy and only actually perform any I/O once `mongoc_cursor_next` is called on the cursor. We wanted our API to not be lazy and perform the commands when their respective helpers were invoked instead, since we figured this was what users were likely expecting to happen. `forceIO` was a parameter you could specify to our internal `MongoCursor` initializer which would force it reach out to the server, but we communicate that now differently via the `MongocCursorWrapper` protocol, removing the need for the parameter. It looks like we just forgot to remove this comment back when those changes were made.

Reply:
Gotcha, thanks!

Reply:
This comment still needs to be removed.